x-pack/elastic-agent/pkg/agent/application/application.go

```diff
@@ -39,10 +39,11 @@ type upgraderControl interface {
 }
 
 // New creates a new Agent and bootstrap the required subsystem.
-func New(log *logger.Logger, pathConfigFile string, reexec reexecManager, statusCtrl status.Controller, uc upgraderControl, agentInfo *info.AgentInfo) (Application, error) {
+func New(log *logger.Logger, reexec reexecManager, statusCtrl status.Controller, uc upgraderControl, agentInfo *info.AgentInfo) (Application, error) {
 	// Load configuration from disk to understand in which mode of operation
 	// we must start the elastic-agent, the mode of operation cannot be changed without restarting the
 	// elastic-agent.
```

Comment:
@michalpristas , based on the comments, it looks like it's expecting the elastic-agent.yml config instead of what is sent (fleet.yml). Replacing it made sure the standalone version gets the right config, also that `mergeFleetConfig` can merge the 2 configs